proposals/0048-otel_delta_temporality_support.md

```diff
@@ -0,0 +1,501 @@
+
+## Implementation Plan
+
+### 1. Experimental feature flag for OTLP delta ingestion
+
+A `--enable-feature=otlp-native-delta-ingestion` was already introduced in https://github.com/prometheus/prometheus/pull/16360, but that doesn't add any type metadata to the ingested series. Update the feature to set the type to gauge and add the `__otel_...` labels.
+
+### 2. Function warnings
+
+Add function warnings when a function is used with series of wrong type or temporality as described in [Function warnings](#function-warnings).
+
+There are already warnings if `rate()`/`increase()` are used without the `__type__="counter"` label: https://github.com/prometheus/prometheus/pull/16632.
+
+### 3. Update documentation
+
+Update documentation explaining new experimental delta functionality, including recommended configs for filtering out delta metrics from scraping and remote write.
+
+### 4. Review deltas in practice and experiment with possible future extensions
+
+Review how deltas work in practice using the current approach, and use experience and feedback to decide whether any of the potential extensions are necessary.
+
+## Potential future extensions
+
+Potential extensions, some may require dedicated proposals.
+
+### Ingest as counter
+
+Instead of ingesting as a Prometheus gauge, ingest as a Prometheus counter. When ingesting a delta metric via the OTLP endpoint, the metric type is set to `counter` / `histogram` (and with PROM-39, the `__type__` label will be `counter` / `histogram`).
+
+This will change the Prometheus definition of a counter (currently: "A counter is a cumulative metric that represents a single monotonically increasing counter whose value can only increase or be reset to zero on restart").
+
+As mentioned in the [Chunks](#chunks) section, `GaugeType` should still be the counter reset hint/header.
+
+Cumulative counter-like metrics ingested via the OTLP endpoint will also have a `__temporality__="cumulative"` label added.
+
+**Pros**
+* Clear distinction between OTEL delta metrics and OTEL gauge metrics when ingested into Prometheus, meaning users know the appropriate functions to use for each type (e.g. `sum_over_time()` is unlikely to be useful for OTEL gauge metrics).
+* Closer match with the OTEL model - in OTEL, counter-like types sum over events over time, with temporality being an property of the type. This is mirrored by having `__type__="counter"` and `__temporality__="delta"/"cumulative"` labels in Prometheus.
+
+**Cons**
+* Introduces additional complexity to the Prometheus data model.
+* Confusing overlap between gauge and delta counters. Essentially deltas already exist in Prometheus as gauges, and deltas can be viewed as a subset of gauges under the Prometheus definition. The same `sum_over_time()` would be used for aggregating these pre-existing deltas-as-gauges and delta counters, creating confusion on why there are two different "types".
```

Comment:
> Essentially deltas already exist in Prometheus as gauges, and deltas can be viewed as a subset of gauges under the Prometheus definition.

Is it really true? What exactly is that subset? That gauge can go down and counter not? Counter can do down meaning reset though. Delta can go down because it's a diff. Gauge is for a current value, Delta is for diff. I see equal confusion when used on either type.